trade.lic

```diff
@@ -0,0 +1,262 @@
+      fput("tell caravan to lead to #{destination}")
+    end
+  end
+
+  def deliver_contract(clerk_id)
+    walk_to(clerk_id)
+    clerk_messages = [
+      '^The \w+ clerk finds everything in order with your merchandise',
+    ]
+    bput("give my contract to clerk", clerk_messages)
+  end
+
+  def pay_dues(clerk_id)
+    echo "Paying dues at clerk_room: #{clerk_id}" if debugging?
+    clerk_messages = [
+        /^You don't have any coins on you to pay dues\./,
+        /^You count out some coins, and the clerk notes that your dues are settled\./,
+        /^You count out some coins, and the clerk notes that your dues are now (.*)\./,
+        /^The clerk says, "You do not owe dues, .*!"/
+    ]
+    walk_to(clerk_id)
+    bput("pay clerk", clerk_messages)
+  end
+
+  def feed_caravan
+    feed_messages = [
+      "The driver takes the feedbag from you",
+      "You offer the .* feedbag to the caravan driver",
+      "The .* sniffs disinterestedly at your .* feedbag",
+      'The .* sticks its nose into your feedbag and munches away happily'
+    ]
+    @equipment_manager.empty_hands
+    bput('remove feedbag', 'You remove')
+    bput("give #{@caravan.noun}", feed_messages)
+    bput('wear feedbag', 'You attach')
+  end
+
+  def load_caravan(clerk_id)
+    echo "Presenting contract to clerk in room: #{clerk_id}" if debugging?
+    walk_to(clerk_id)
+    give_clerk(clerk_id)
+  end
+
+  def get_new_contract(minister_id)
+    echo "Getting a new contract from room #{minister_id}" if debugging?
+    walk_to(minister_id)
+    minister_messages = [
+      "The minister plucks a contract from the hands",
+      "The minister reminds you that you still owe .*" 
+    ]
+    response = bput("ask minister for contract", minister_messages)
+    case response  
+    when /The minister reminds you/
+      /The minister reminds you that you still owe (?<dues_amt>.*) (?<dues_currency>Kronars|Lirums|Dokoras) in dues to (?:The )?(?<outpost>.*) and asks that you pay the clerk before s?he issues you a new contract\./ =~ response
+      echo "Need to pay #{dues_amt} #{dues_currency} to #{outpost}" if debugging?
+      ensure_copper_on_hand(dues_amt.to_i, @hometown)
+      pay_dues(@town_data[outpost]['shipment_clerk']['id'])
+      get_new_contract(minister_id)
+    end
+  end
+
+  def caravan_present?
+    return unless @caravan.adjective
+    return unless @caravan.noun
+    return DRRoom.room_objs.find{ |obj| obj =~ /#{@caravan.adjective} #{@caravan.noun}/}
+  end
+
+  def give_clerk(clerk_id)
+    walk_to(clerk_id)
+    bput("get my contract", 'You get')
+    Flags.add('caravan-upgrade', /Unfortunately, this load would kill/)
+    bput("give my contract to clerk", /The .* clerk accepts your contract and peruses it\./)
+    recall_caravan if Flags['caravan-upgrade']
+    bput("put my contract in my #{@contract_container}", 'You put')
+    Flags.delete('caravan-upgrade')
+  end
+end
+
+trade = Trade.new
+loop { trade.run }
```

Comment:
Generally scripts have been designed to run once, with a second script responsible for looping (and stopping, etc). I'm not saying you have to change anything, I'm just not sure it would be intuitive that this loops indefinitely.

Reply:
I was originally going to add a TradingBuddy class to this that would manage the start stop based on some user preferences, inspired by hunting buddy. I'll remove this loop and let a future commit handle the looping